### PR TITLE
Feature/to do

### LINKED ISSUE #5: Todo List (bottom-right)
This feature (pop-up) will query the user for new todo items and organize them in a list by addition time order. The user has the option to mark a completed todo and/or delete any todo item from the list. When the list is complete, a congratulations message is shown.

Tasks:
- [ ] Generate feature HTML;
- [ ] Add functionality with jQuery;
- [ ] Design (CSS).

(issue configuration in progress)

### PR DESCRIPTION
**Pull request by jrpcoder**
####  Checklist

- [x] Pull request is from a branch (not master)
- [ ] Pull request contains only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Closes open issue : closes #5 

#### Description

- todo list feature is complete. Functionality is identical to the real feature.
- when more than 20 todos are added, a scroll is generated.
- if the todos overflow the div element, ellipsis are added to the todos limit with the div. This way, the structure is mantained.
- the feature accounts for the amount of todos completed and displays the info. when all todos are deleted, the feature resolves to the initial form with the smile.
- the data is saved to localStorage and the todos kept on an array.
---------------------
additionally, I corrected the focustodo delete button positioning, as well as the quote feature positioning. which was giving problems for larger quotes.  
Note: Please, feel free to squash commits and delete branch, if the PR is approved. Thanks!

### CODE REVIEW

File: src/components/toDo/ToDoList.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      amountUndoneToDos: this.howManyUndoneTodos(updatedDoneTodosArr),
+    });
+  }
+
+  toggleModal() {
+    if (this.state.toggleModal) {
+      this.setState({
+        toggleModal: !this.state.toggleModal,
+      });
+    } else {
```
Comment (marigerr) [inline src/components/toDo/ToDoList.jsx]: This if/else block seems unnecessary?  Wouldn't this work?

```
  toggleModal() {
    this.setState({
      toggleModal: !this.state.toggleModal,
    });
  }
```

File: src/assets/css/listToDoItem.css [hunk 2]
```diff
@@ -49,16 +86,33 @@
...
   height: 30px;
   line-height: 30px;
-  font-size: 20px;
 }
 
-.deleteButton > .letterX {
-  vertical-align: middle;
-  height: 30px;
-  line-height: 28px;
+.letterX-todoList {
```
Comment (marigerr) [inline src/assets/css/listToDoItem.css]: The focustodo delete button still looks un-centered on my screen?

File: src/components/toDo/todoModal.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  crossToDo(todo) {
+    if (todo.toDoDone) {
+      return this.props.toDoStyle;
+    }
+    return 'todoItemList';
+  }
+
+  // iterate through the todos array and generate jsx to render the todos list
+  toDoList(toDoArr) {
+    const { toDoListArr } = toDoArr;
```
[the remark this answers is not in the document]
Comment (jrpcoder) [reply src/components/toDo/todoModal.jsx]: Thanks! I was unsure at the time, but I can see how this is more readable.

File: src/components/toDo/ToDoList.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        toDoDone: false,
+        id: getCurrentTime(),
+      };
+      addToLocalStorageArray('toDosArr', newToDoObj);
+      const updatedToDosArr = getFromLocalStorage('toDosArr');
+      this.setState({
+        toDosArr: updatedToDosArr,
+        amountUndoneToDos: this.howManyUndoneTodos(updatedToDosArr),
+        todoInput: '',
+      }, () => { console.log(this.state); });
```
[the remark this answers is not in the document]
Comment (jrpcoder) [reply src/components/toDo/ToDoList.jsx]: completely forgot to remove it :) thanks

File: src/components/toDo/ToDoList.jsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  showDeleteButton() {
+    return <div className="deleteButton-todoList"><span className="letterX-todoList">&times;</span></div>;
+  }
+
+  // update the array containing the todos list by deleting the selected todo
+  deleteTodo(todoId) {
+    removeFromLocalStorageArray('toDosArr', 'id', todoId);
+    const updatedToDosArr = getFromLocalStorage('toDosArr');
```
[the remark this answers is not in the document]
Comment (jrpcoder) [reply src/components/toDo/ToDoList.jsx]: Great suggestion! I'll rearrange it.
what do you think about doing the same on the addToLocalStorageArray also?

File: src/components/toDo/ToDoList.jsx (hunk 1 above)
Comment (jrpcoder) [reply src/components/toDo/ToDoList.jsx]: Ofc your suggestion works. Doing a conditional for the exact same code is totally unnecessary. Thanks for noticing it.

File: src/components/toDo/ToDoList.jsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    return (
+      <div className="todo-list-container">
+        {this.state.toggleModal && <ToDoModal
+          deleteTodo={todoId => this.deleteTodo(todoId)}
+          clickCheckbox={todoId => this.clickCheckbox(todoId)}
+          containerStyle={this.state.containerStyle}
+          toDoStyle={this.state.toDoStyle}
+          deleteButtonStyle={() => this.showDeleteButton()}
+          checkboxType={toDoDone => this.showCheckbox(toDoDone)}
+          toDoListArr={this.state.toDosArr}
```
[the remark this answers is not in the document]
Comment (jrpcoder) [reply src/components/toDo/ToDoList.jsx]: I agree. changed.

File: src/assets/css/listToDoItem.css (hunk 2 above)
Comment (jrpcoder) [reply src/assets/css/listToDoItem.css]: really? Is it just for the times symbol? thanks

File: src/components/toDo/ToDoList.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (jrpcoder) [reply src/components/toDo/ToDoList.jsx]: Ok, I went ahead and changed the addtolocalstoragearray also ;)

[the remark this answers is not in the document]
Comment (marigerr) [reply src/components/toDo/ToDoList.jsx]: Ah yes, I think that would work there too.   I use both of those utility functions elsewhere but doesn't appear to cause any problems if I don't assign the return value anywhere.

I tested changing `addToLocalStorageArray`.   I had to rewrite a couple lines, but this seems to work.

```
function addToLocalStorageArray(localStorageKey, objectToAdd) {
  let storageArray = [];
  if (localStorageKeyExists(localStorageKey)) {
    storageArray = getFromLocalStorage(localStorageKey);
    console.log(storageArray);
  }
  storageArray.push(objectToAdd);
  addToLocalStorage(localStorageKey, storageArray);
  return storageArray;
}
```

File: src/assets/css/listToDoItem.css (hunk 2 above)
Comment (marigerr) [reply src/assets/css/listToDoItem.css]: yes sorry!
It is off for both plus and times symbol on the main focus todo.   
Its not off by very much, but definitely is not centered.

image

image

File: src/components/toDo/ToDoList.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (marigerr) [reply src/components/toDo/ToDoList.jsx]: woops -  edited out console.log from my suggested code for `addToLocalStorageArray`

**Comment by marigerr** [comment]
There are a couple more console.logs that would be great to comment out/delete, if you don't mind.  Thanks!

- focusToDo.jsx_  line 45
- Quote.jsx  line 29 and line 36
- App.jsx line 46 and line 47
- likeheart.jsx line 45 and line 48

File: src/assets/css/listToDoItem.css (hunk 2 above)
Comment (jrpcoder) [reply src/assets/css/listToDoItem.css]: Thanks! added a possible fix. How does it look like?

**Pull request merged**